ipywidgets/widgets/widget.py

```diff
@@ -759,3 +759,23 @@ def _gen_repr_from_keys(self, keys):
             for key in keys
         )
         return '%s(%s)' % (class_name, signature)
+
+
+    def __copy__(self):
+
+        cls = self.__class__
+        result = cls.__new__(cls)
+        result.__init__()
+
+        new_state = {key: value for key, value in self.get_state().items() if
+                     not key.startswith('_') and
+                     not key == 'layout' and
+                     not key == 'style'}
```

Comment:
Maybe the state could be directly passed to `__init__` as kwargs.

Also, rather than overriding the method for each widget type that may have subwidget, a good way to exclude things may be to exclude attributes that are of type Widget?

A list of synchronized keys is available in the `keys` attribute.

Finally, I don't quite know if there may be other state in the object that is no synchronized with the front-end, and would need to be included in a copy operator...